custom/inddex/fixtures.py

```diff
@@ -121,13 +121,20 @@ def foods(self):
     def foods_by_name(self):
         return {food.food_name: food for food in self.foods.values()}
 
+    @cached_property
+    def _nutrients(self):
+        return [
+            Nutrient(**item_dict)
+            for item_dict in self._get_fixture_dicts('nutrients_lookup')
+        ]
+
+    @cached_property
+    def nutrient_names(self):
+        return [n.nutrient_name_unit for n in self._nutrients]
```

Comment:
There aren't a lot of nutrients, and this code isn't used a lot (so I'm not suggesting you change this). In the past I've wondered how to decide when to cache, and when to calculate. e.g. `_nutrients`, `nutrient_names` and `_nutrient_names_by_code` are all related. Under what conditions would you write `nutrient_names` as just, say, 
```python
    @property
    def nutrient_names(self):
        return self._nutrient_names_by_code.values()
```
... or if you're balancing speed vs memory use, and this uses so little memory, it's better just to cache?

Reply:
Oh yeah - I meant to look at that later, but never did.  IMO parsing and wrapping the fixture elements _does_ needs to be cached for sure.  Beyond that, I originally wrote this like you suggested, but later realized that I need to preserve the order of fixtures (the couch view returns them [in order](https://github.com/dimagi/commcare-hq/blob/c320b77efbc5f1ca5aa410c0f851278e983f657b/corehq/apps/fixtures/_design/views/data_items_by_domain_type/map.js#L3-L3)).

`nutrient_names` is only used twice, and I don't think it's expensive to calculate, given `_nutrients`, so it'd probably be fine to leave as a normal `@property`.  `_nutrient_names_by_code` is only used once - while constructing `food_compositions`.  The current usage is sub-optimal, as it only needs to exist for the duration of that method call, but as-implemented, it's cached for the lifetime of the `FixtureAccessor` instance.  For that matter, the current lifecycle of `_nutrients` exceeds its utility by the same logic.

Here's an alternate approach that feels dirty, but is perhaps more optimized?  IDK, not sure I'm a fan.
```python
    @cached_property
    def _populate_nutrients(self):
        nutrients = [
            Nutrient(**item_dict)
            for item_dict in self._get_fixture_dicts('nutrients_lookup')
        ]
        self._nutrient_names = [n.nutrient_name_unit for n in nutrients]
        self._nutrient_names_by_code = {n.nutrient_code: n.nutrient_name_unit
                                        for n in nutrients}

    @property
    def nutrient_names(self):
        self._populate_nutrients()
        return self._nutrient_names

    @property
    def nutrient_names_by_code(self):
        self._populate_nutrients()
        return self._nutrient_names_by_code
```

Reply:
Yeah, not a fan either. I reckon it's better as it is.